pkg/helm/helm.go

```diff
@@ -87,14 +87,11 @@ func (h *Hydrator) templateArgs(ctx context.Context, destDir string) ([]string,
 }
 
 func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
-	var values map[string]interface{}
-	err := json.Unmarshal([]byte(h.Values), &values)
-	if err != nil {
-		return []string{}, fmt.Errorf("failed to unmarshal helm.values, error: %w", err)
-	}
-	for key, val := range values {
-		args = append(args, "--set", fmt.Sprintf("%s=%v", key, val))
+	valuesPath := "tmp/chart-values.yaml"
```

Comment:
Instead of hard-coding a temp file, could we use `os.CreateTemp` to generate a temp file for us instead? This would allow Golang to determine the best directory to place the temp file as well as create globally unique filenames. I'm not sure if we need to worry about other processes writing to this file as well?

Reply:
The reason to have a hard-coding temp file is that each helm sync container only sync one chart, and only need one values file. If retry/re-sync needed, the same values file should be overwrite instead of creating new temp file.

Reply:
Instead of using a temp file, can we generate the values file under the current chart folder? With this approach, the entire folder will be removed when retrying or resyncing from a different chart.

If we go with the temp file, at least let golang decide which is the temp directory, `os.TempDir()`.

Reply:
Change to use os.TempDir() instead of using /tmp directly. Since we are using helm template instead of download the chart folder first and do the rendering later, this values file can not be put under chart folder. So will go with temp file